### ISSUE TITLE
checkpoint

### ISSUE THREAD

**Issue by Jx-Tan** [282857341/nnFormer#36]
Hi @282857341 

Because my GPU memory is small, so I use the checkpoint in Swin_Unet_1_gelunorm.py----BasicLayer(nn.Module)---`x = checkpoint.checkpoint(blk, x , attn_mask)`, and `use_checkpoint=True`. After that, I can train it, but in epoch: 0, I get a UserWarning: None of the inputs have requires_grad=True. Gradients will be None          
warnings.warn("None of the inputs have requires_grad=True. Gradients will be None

But in epoch 1, 2, 3 ...   I did not see this warning again. Does this Warning affect the training process? And is it affect the accuracy of model?

**Comment by 282857341** [comment]
Sorry for late reply.
use_checkpoint=True may still exits the bugs, and I don't recommend to use it.

**Comment by 282857341** [comment]
if the gpu memory problem still exits, you can reduce the embedding dim or depths
https://github.com/282857341/nnFormer/blob/1ebde811af96715e40fcf850d190cfbef6093ffe/nnformer/network_architecture/Swin_Unet_s_ACDC_2laterdown.py#L891-L895
